src/substrate/poller.ts

```diff
@@ -47,8 +47,13 @@ export class Poller extends IEventPoller<ApiPromise, Block> {
     const blockNumbers = [ ...Array(range.endBlock - range.startBlock).keys()]
       .map((i) => range.startBlock + i);
     log.debug(`Fetching hashes for blocks: ${JSON.stringify(blockNumbers)}`);
-    const hashes: Hash[] = await this._api.query.system.blockHash.multi(blockNumbers);
 
+    // the hashes are pruned when using api.query.system.blockHash.multi
+    // therefore fetching hashes from chain. the downside is that for every
+    // single hash a separate request is made
+    const hashes = await Promise.all(
+        blockNumbers.map( async  (number)=> (await this._api.rpc.chain.getBlockHash(number))))
```

Comment:
Inner async/await is unnecessary here -- `rpc.chain.getBlockHash()` already returns a promise, so you can just return that and await the `Promise.all` result.

Reply:
thanks for pointing this out. resolved in latest commit.